python/teams_ai/planner/planner.py

```diff
@@ -0,0 +1,40 @@
+"""
+Copyright (c) Microsoft Corporation. All rights reserved.
+Licensed under the MIT License.
+"""
+
+from abc import ABC, abstractclassmethod
+from typing import Any, Callable, Union
+from teams_ai.turn_state import TurnState
+
+class Planner(ABC):
+    """
+    interface implemented by all planners
+    """
+
+    @abstractclassmethod
+    async def generate_plan(self, turn_context:TurnContext, state: TurnState, prompt_name_or_template: Union[str, PromptTemplate], *, history_options: AIHistoryOptions) -> Plan:
```

Comment:
according to #206, the parameters should be turn_context, state, & ai

Reply:
For `prompt_name_or_template`, I feel the prompt name / template is required, otherwise we won't know which prompt to use when generating the plan.
For `history_option`, it should be better to only accept options that is valid to the function. Otherwise, it could be hard to call this function because people don't know which options is required and need to initialize the whole AI instance, which is not necessary. Please let me know if there're other options available to the planner.